app/javascript/components/Course/GithubRepos/CourseGithubRepoProjectReposStatistics.jsx

```diff
@@ -0,0 +1,524 @@
+                this.computeOverallStudentStats();
+            }
+
+            let setIssueTimelineItems = (o) => {
+                let new_timeline_query_results = this.state.timeline_query_results;
+                let new_timeline_stats = this.state.timeline_stats;
+                let new_timeline_combined_count = this.state.timeline_combined_count;
+                let new_timeline_combined_count_by_student = this.state.timeline_combined_count_by_student;
+
+                let new_overall_combined_count_by_team = 
+                        this.state.overall_combined_count_by_team;
+                let new_errors = this.state.errors;
+
+                if(o.success) {
+                    try {
+                        new_timeline_query_results[this.repoName(repo)] = cloneDeep(o);
+                        let this_repos_stats = IssueTimelineItems.computeStats(o.data, this.props.databaseId_to_team);
+                        new_timeline_stats[this.repoName(repo)] = this_repos_stats;
+                        new_timeline_combined_count = combineCounts(
+                            new_timeline_combined_count,
+                            this_repos_stats.statistics.timelineItemsTeamsCount
+                        );
+                        new_timeline_combined_count_by_student = combineCounts(
+                            new_timeline_combined_count_by_student,
+                            this_repos_stats.statistics.timelineItemsCounts
+                        );
+                        new_overall_combined_count_by_team = combineCounts(
+                            new_overall_combined_count_by_team,
+                            this_repos_stats.statistics.timelineItemsTeamsCount
+                        );
+                    } catch (e) {
+                        new_errors[this.repoName(repo)].push(errorToObject(e))
+                    }
+                } else {
+                    new_errors[this.repoName(repo)].push(o.error)
+                }
+                    
+                this.setState({ 
+                    timeline_query_results: new_timeline_query_results,
+                    timeline_stats: new_timeline_stats,
+                    timeline_combined_count: new_timeline_combined_count,
+                    timeline_combined_count_by_student: new_timeline_combined_count_by_student,
+                    overall_combined_count_by_team: new_overall_combined_count_by_team,
+                    errors: new_errors,
+                });
+                this.computeOverallTeamStats();
+                this.computeOverallStudentStats();
+            }
+
+            let issueEditsQueryObject = new GraphqlQuery(url,ieQuery,ieAccept,setIssueEdits,{repo: repo.name});
+            issueEditsQueryObject.post();
+            let timelineItemsQueryObject = new GraphqlQuery(url,tlQuery,tlAccept,setIssueTimelineItems,{repo: repo.name});
+            timelineItemsQueryObject.post();
+        });
+    }
+
```

Comment:
Definitely feels like some of this (and this file in general) could be broken up into smaller files. That being said, if this logic is only going to be used in this component, then I can understand it as a necessary evil.